pkg/scheduler/queue/queue.go

```diff
@@ -291,6 +302,29 @@ func (q *RequestQueue) trySendNextRequestForQuerier(call *nextRequestForQuerierC
 		return false
 	}
 
+	{
+		// temporary observation of query component load balancing behavior before full implementation
+		schedulerRequest, ok := req.req.(*SchedulerRequest)
+		if ok {
+			queryComponentName := QueryComponentNameForRequest(schedulerRequest)
+			exceedsCapacity, queryComponent := q.QueryComponentCapacity.ExceedsCapacityForComponentName(
+				queryComponentName,
+				q.connectedQuerierWorkers,
+				q.queueBroker.tenantQueuesTree.ItemCount(),
+				q.waitingNextRequestForQuerierCalls.Len(),
+			)
+
+			if exceedsCapacity {
+				level.Warn(q.log).Log(
+					"msg", "querier worker connections in use by query component exceed reserve capacity",
```

Comment:
Should we make this clearer that this is just for testing and the query-scheduler didn't do anything different as a result of this situation?

(I'm imagining someone being in the heat of an incident, seeing this message and going off on a wild goose chase, not realising this is just for evaluation purposes.)

Given this, we should probably also drop the severity down to debug as well.

Reply:
updated the message and changed to Info level. We don't want debug level because then we have to turn debug logging on to observe it. Info works because it's not warn or error, which would be filtered for when investigating an incident.